databases/functions/fcmToken/findAllTokenByUserId.js

```diff
@@ -1,16 +1,23 @@
+// eslint-disable-next-line no-unused-vars
 const {Model} = require('sequelize');
+const {sequelize} = require('../../models');
 
 /**
  *
  * @param {Model} fcmTokenModel
  * @param {String} userId
  */
 module.exports = async (fcmTokenModel, userId) => {
-  const userTargetTokens = await fcmTokenModel.findAll({
-    where: {
-      user_id: userId
+  const userTargetTokens = sequelize.query(
+    'SELECT DISTINCT(token) as token from user_token WHERE user_id = :userId',
+    {
+      replacements: {
+        userId
+      },
+      raw: true,
+      type: sequelize.QueryTypes.SELECT
     }
-  });
+  );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new implementation uses a raw SQL query to fetch distinct tokens for a user. While this approach is more efficient than the previous one, it lacks error handling. If the query fails due to any reason (like database connection issues), the application might crash or behave unexpectedly. Consider adding a try-catch block to handle potential errors.

```diff
 module.exports = async (fcmTokenModel, userId) => {
+   try {
    const userTargetTokens = sequelize.query(
      'SELECT DISTINCT(token) as token from user_token WHERE user_id = :userId',
      {
        replacements: {
          userId
        },
        raw: true,
        type: sequelize.QueryTypes.SELECT
       }
    );
 
    return userTargetTokens;
+   } catch (error) {
+     console.error(`Failed to fetch tokens for user ${userId}: `, error);
+     throw error;
+   }
};
```
This way, if an error occurs during the execution of the query, it will be caught and logged, and then re-thrown to be handled by the caller. This improves the robustness of your code and makes debugging easier in case of issues.


<!-- This is an auto-generated comment by OSS CodeRabbit -->